packages/connectivity/src/scp-cf/destination/destination-service.ts

```diff
@@ -323,6 +328,42 @@ function errorMessageFromResponse(
     : '';
 }
 
+function retryDestination(
+  destinationName: string
+): Middleware<AxiosResponse<DestinationJson>, HttpMiddlewareContext> {
+  return options => () => {
+    let retryCount = 0;
+    return asyncRetry.default(
+      async bail => {
+        try {
+          const destination = await options.fn();
+          if (retryCount < 3) {
```

Comment:
[q] Is this if necessary? the retry count will never be more than 3 right?

Reply:
The reason for this is that we return the destination with the errors in the auth tokens. I am not sure if this is a good idea because in most cases the destintaion will be useless. However, this is the way it was before and I wanted to keep this. Hence the `buildAuthorizationHeaders` is not executed on the last retry to make the promise resolve.